go-controller/pkg/clustermanager/network_cluster_controller.go

```diff
@@ -67,10 +70,21 @@ type networkClusterController struct {
 	// event recorder used to post events to k8s
 	recorder record.EventRecorder
 
+	statusReporter NetworkStatusReporter
+
+	// nodeName: errMessage
+	nodeErrors     map[string]string
```

Comment:
Would you see udefulness in reflecting in a TODO comment some of the conversation that we had about caching the errors vs being dumber here but smarter on when this errors get reported trhough a msarter retry mechanism?

Reply:
yes! added a link to our conversation here to the "updateNetworkStatus" function comment